erts/emulator/beam/instrs.tab

```diff
@@ -888,12 +888,16 @@ catch_end(Y) {
 try_end(Y) {
     c_p->catches--;
     make_blank($Y);
-    if (is_non_value(r(0))) {
-        c_p->fvalue = NIL;
-        r(0) = x(1);
-        x(1) = x(2);
-        x(2) = x(3);
-    }
+}
+
+try_case(Y) {
+    c_p->catches--;
+    make_blank($Y);
+    ASSERT(is_non_value(r(0)));
+    c_p->fvalue = NIL;
+    r(0) = x(1);
+    x(1) = x(2);
+    x(2) = x(3);
```

Comment:
maybe rewrite as

    try_case(Y) {
      $try_end($Y);
      ASSERT(is_non_value(r(0)));
      ....

and the same for catch_end.

Reply:
Good idea!